unit-tests/py/rspy/libci.py

```diff
@@ -146,42 +146,56 @@ def __init__( self, source, line_prefix, context ):
         self.derive_tags_from_path( source )
 
     def derive_config_from_text( self, source, line_prefix ):
-        regex = r'^' + line_prefix + r'(\S+)((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
-        for context in file.grep( regex, source ):
-            match = context['match']
+        regex = r'^' + line_prefix + r'([^\s:]+)(:(\S+))?((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
+        for line in file.grep( regex, source ):
+            match = line['match']
             directive = match.group( 1 )
-            text_params = match.group( 2 ).strip()
+            directive_context = match.group( 3 )
+            text_params = match.group( 4 ).strip()
             params = [s for s in text_params.split()]
-            comment = match.group( 3 )
+            comment = match.group( 5 )
+            if directive_context:
+                if not directive_context.startswith('!') and directive_context != self.context:
```

Comment:
```
With !        directive_ctx == context   |   RESULT
--------      ------------------------   |   --------
      0                     0            |     IGNORE
      0                     1            |     USE
      1                     0            |     USE
      1                     1            |     IGNORE
```
So we want to ignore when `directive_context.startswith('!') == (directive_context == self.context)`

Reply:
Should I maybe add a comment with the table?
Also was thinking of adding a comment explaining the regex, thoughts?

Reply:
You can, feel free.

Reply:
The regex can be explained but it's easier to split the line into multiple parts:
```
regex = r'^' + line_prefix
regex +=r'([^\s:]+)'           # 1: directive
regex += r'(:(\S+))?'          # 3: optional context
regex += r'((?:\s+\S+)*?)'     # 4: params
regex += r'\s*(?:#\s*(.*))?$'  # 5: optional comment
```
BTW, the 2nd match can be removed

Reply:
How can I remove the second match without changing the regex or the third match?

Reply:
Instead of `(:(` use `(?::(`

Reply:
`(?:` means "start a non-capturing group"